File: tests/fixtures/sample.ts
// Sample TypeScript file for testing tree-sitter parsing.

import { Something } from "./module";
import * as fs from "fs";

interface Greeter {
  greet(name: string): string;
}

abstract class BaseGreeter implements Greeter {
  abstract greet(name: string): string;
}

class HelloGreeter extends BaseGreeter {
  private prefix: string;

  constructor(prefix: string = "Hello") {
    super();
    this.prefix = prefix;
  }

  greet(name: string): string {
    return `${this.prefix}, ${name}!`;
  }

  private helper(): void {
    console.log("helper");
  }
}

function processData(data: number[]): number {
  let sum = 0;
  for (const v of data) {
    if (v > 0) {
      for (let i = 0; i < v; i++) {
        sum += i;
      }
    }
  }
  return sum;
}

const arrowFunc = (x: number): number => x * 2;

export { HelloGreeter, processData };

File: tests/fixtures/Sample.java
// Sample Java file for testing tree-sitter parsing.
package com.example;

import java.util.List;
import java.util.ArrayList;

public interface Greeter {
    String greet(String name);
}

public abstract class BaseGreeter implements Greeter {
    protected String prefix;

    public abstract String greet(String name);
}

public class HelloGreeter extends BaseGreeter {
    public HelloGreeter(String prefix) {
        this.prefix = prefix;
    }

    @Override
    public String greet(String name) {
        return String.format("%s, %s!", prefix, name);
    }

    private void helper() {
        System.out.println("helper");
    }
}

public class DataProcessor {
    public int processData(List<Integer> data) {
        int sum = 0;
        for (int v : data) {
            if (v > 0) {
                for (int i = 0; i < v; i++) {
                    sum += i;
                }
            }
        }
        return sum;
    }

    public static void main(String[] args) {
        HelloGreeter g = new HelloGreeter("Hello");
        System.out.println(g.greet("World"));
    }
}
